internal/middleware/ratelimiter.go

```diff
@@ -87,10 +89,20 @@ func RateLimiter(client *db.DiceDB, next http.Handler, limit, window int) http.H
 			}
 		}
 
+		// Add rate limit headers
+		addRateLimitHeaders(w, limit, limit-(requestCount+1), requestCount+1, currentWindow+int64(window))
```

Comment:
Wouldn't there be a timelapse between setting headers here and DB fetch and response, due to network latency etc.? May be we can set the headers at very end before sending response.

Please analyse this once.